File: src/Modules/Uniforms/Domain/Entities/WorkOrder.cs
using SharedKernel.Domain;

namespace Modules.Uniforms.Domain.Entities;

public class WorkOrder : TenantScopedEntity, IAuditableEntity
{
    public string OrderNumber { get; set; } = string.Empty;
    public Guid CustomerId { get; set; }
    public string CustomerName { get; set; } = string.Empty;
    public string? CustomerPhone { get; set; }
    public string? CustomerEmail { get; set; }
    public string Status { get; set; } = "draft";
    public DateTime? DueDate { get; set; }
    public string? Notes { get; set; }
    public decimal TotalAmount { get; set; }
    public Guid CreatedBy { get; set; }
    public Guid? UpdatedBy { get; set; }

    public ICollection<WorkOrderLine> Lines { get; set; } = new List<WorkOrderLine>();
}

File: frontend/shared-types/src/work-order.ts
import { z } from 'zod';

export interface WorkOrder {
  id: string;
  orderNumber: string;
  customerId: string;
  customerName: string;
  customerPhone: string | null;
  status: string;
  dueDate: string | null;
  totalAmount: number;
  lineCount: number;
  notes: string | null;
  createdAt: string;
}

export interface WorkOrderDetail {
  id: string;
  orderNumber: string;
  customerId: string;
  customerName: string;
  customerPhone: string | null;
  customerEmail: string | null;
  status: string;
  dueDate: string | null;
  totalAmount: number;
  notes: string | null;
  createdAt: string;
  lines: WorkOrderLine[];
}

export interface WorkOrderLine {
  id: string;
  description: string;
  productVariantId: string | null;
  productName: string | null;
  sku: string | null;
  quantity: number;
  unitPrice: number;
  lineTotal: number;
}

export const CreateWorkOrderSchema = z.object({
  customerId: z.string().min(1),
  customerName: z.string().min(1),
  customerPhone: z.string().nullish(),
  customerEmail: z.string().nullish(),
  dueDate: z.string().nullish(),
  notes: z.string().nullish(),
  lines: z.array(z.object({
    description: z.string().min(1, 'Description is required'),
    productVariantId: z.string().nullish(),
    productName: z.string().nullish(),
    sku: z.string().nullish(),
    quantity: z.number().int().positive('Quantity must be positive'),
    unitPrice: z.number().min(0, 'Unit price must be non-negative'),
  })).min(1, 'At least one line item is required'),
});

export type CreateWorkOrderRequest = z.infer<typeof CreateWorkOrderSchema>;
